### PR TITLE
Build docker image from this sample

### PR DESCRIPTION
**Pull request by saragluna**
## Purpose
This PR will add the following features to the sample project:
- Add Dockerfile to build a docker image
- Add a GitHub action to automate the docker image build and publish
- Improve the '/' path to not return 404 when visiting the application's root path

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: .github/workflows/docker-image.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+name: Docker Image CI
+
+on:
+  push:
+    tags:
+    - 'v[0-9]+.[0-9]+.[0-9]+'
+    
+env: 
+  IMAGE_NAME: samples/event-driven
```
Comment (stliu) [inline .github/workflows/docker-image.yml:9]: `asa-samples/event-driven` ?

File: .github/workflows/docker-image.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        distribution: 'temurin'
+        cache: 'maven'
+    - name: Build with Maven
+      run: ./mvnw clean package  
+    - name: Get current git tag
+      id: tag
+      uses: devops-actions/action-get-tag@v1.0.2
+      with:
+        strip_v: true
+    - name: Set up Docker Buildx
```
Comment (stliu) [inline .github/workflows/docker-image.yml:30]: typo or is it just *buildx* ?

Comment (saragluna) [reply .github/workflows/docker-image.yml:30]: not typo

File: .github/workflows/docker-image.yml (hunk 1 above)
Comment (saragluna) [reply .github/workflows/docker-image.yml:9]: the image name will be like `unlisted/azurespringapps/samples/hello-world` in mcr, will it be weird if we call it `unlisted/azurespringapps/asa-samples/hello-world`?

**Review by stliu — APPROVED** [review]

**Pull request merged**